app/Http/Requests/StoreOfferRequestForm.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Models\Offer;
+use App\Rules\AvailableStoreRule;
+use App\Services\UrlService;
+use Closure;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class StoreOfferRequestForm extends AbstractRequestForm
+{
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'url' => [
+                'required',
+                'url',
+                new AvailableStoreRule,
+            ],
+            'name' => ['required', 'string', 'min:3', 'max:255'],
+            'is_active' => ['required', 'int', Rule::in([0, 1])]
+        ];
+    }
+
+    public function store(): Offer
+    {
+        $data = $this->validated();
+        $data['domain'] = UrlService::getDomain($data['url']);
```

Comment:
Najs ale nie lepiej to zrobić dynamicznie niz zapisywać w bazie, teraz jesli zmieni się url to musisz zaktualizować domenę, a bez zapisywania tego mogłeś zrobić akcesor który zwróci domenę na podstawie aktualnego url?

Reply:
Nie chcę pozwalać na aktualizację url, muszę tylko zmodyfikować metodę zmieniającą required na sometimes. Dodatkowo chciałbym po domenie filtrować listę, a jeśli zostawię to jako akcesor, to wtedy to filtrowanie będzie mniej optymalne? Tzn. wiem że tutaj to bez znaczenia, ale jak do tego podejść w większym projekcie?